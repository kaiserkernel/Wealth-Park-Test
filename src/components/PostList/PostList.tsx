/* eslint-disable @typescript-eslint/no-unnecessary-condition */
import { FC } from 'react'

import { IPostList } from '@src/types'
import Post from '../PostListCard/PostListCard'

const PostList: FC<IPostList> = ({ data }) => {
  if (!data?.pages?.length || data?.pages?.length === 0) {
    return null
  }
  return (
    <div data-testid="post-list" className="flex flex-col justify-center">
      {data.pages.map((results) => results.map((post) => <Post key={post.id} post={post} />))}
    </div>
  )
}

export default PostList
