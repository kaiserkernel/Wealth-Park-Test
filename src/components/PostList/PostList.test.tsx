import { render, screen } from '@testing-library/react'

import PostList from './PostList'

describe('Post List', () => {
  const DATA = {
    data: { pages: [] },
  }

  it('renders correctly', () => {
    const { container } = render(<PostList {...DATA} />)
    expect(container).toMatchSnapshot()
  })

  it('renders PostList', () => {
    render(<PostList {...DATA} />)

    const postList = screen.getByTestId('post-list')

    expect(postList).toBeInTheDocument()
  })
})
