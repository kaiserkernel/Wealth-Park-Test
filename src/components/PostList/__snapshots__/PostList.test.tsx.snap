// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post List renders correctly 1`] = `
<div>
  <div
    class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 xl:grid-cols-4 gap-8"
    data-testid="post-list"
  />
</div>
`;
